### PR TITLE
GUVNOR-2873: Guided Decision Table Editor: Change Edit, Insert and View menus to use MenuItems

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2873

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:-

- https://github.com/uberfire/uberfire/pull/617
- https://github.com/droolsjbpm/drools-wb/pull/394

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java [hunk 1]
```diff
@@ -18,45 +18,96 @@
...
-            element.removeClassName( style.getCssName() );
+        if (style != null && style.getCssName() != null && !style.getCssName().isEmpty()) {
+            element.removeClassName(style.getCssName());
         }
     }
 
+    // --------------------------------
+    // Helpers for errai-ui HTMLElement
+    // --------------------------------
+
```
Comment (csadilek) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java]: If @mbarkley agrees those would make a great enhancement to Errai. Then they can easily be reused everywhere.

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/menu/MenuItemDividerView.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * limitations under the License.
+ */
+package org.uberfire.ext.widgets.common.client.menu;
+
+import javax.enterprise.context.Dependent;
+
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Templated
+@Dependent
```
Comment (csadilek) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/menu/MenuItemDividerView.java]: Don't need ```@Dependent``` as ```@Templated``` is implicitly dependent-scoped.

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/menu/MenuItemWithIconView.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import org.jboss.errai.common.client.dom.Div;
+import org.jboss.errai.common.client.dom.ListItem;
+import org.jboss.errai.common.client.dom.Span;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.uberfire.ext.widgets.common.client.common.StyleHelper;
+
+@Templated
+@Dependent
```
Comment (csadilek) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/menu/MenuItemWithIconView.java]: Don't need Dependent.

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java (hunk 1 above)
Comment (mbarkley) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java]: +1

Comment (manstis) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java]: @mbarkley Where would you like me to put it? In the nicest of ways ;-)

**Comment by manstis** [comment]
@csadilek Modifications made as suggested (other than moving ```StyleHelper``` to ```errai```). This class (without the ```HTMLElement``` changes is used elsewhere in the KIE codebase so I'd rather move as a separate activity, if OK?)

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java (hunk 1 above)
Comment (mbarkley) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java]: @manstis I think adding all the StyleHelper methods into [DOMUtil](https://github.com/errai/errai/blob/master/errai-common/src/main/java/org/jboss/errai/common/client/dom/DOMUtil.java) would be fine.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by csadilek** [comment]
jenkins retest this.

File: uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java (hunk 1 above)
Comment (manstis) [inline uberfire-extensions/uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/StyleHelper.java]: @mbarkley @csadilek See https://github.com/errai/errai/pull/232

**Pull request merged**